### DISCUSSION TITLE
Option to control volume per app with shortcuts or something

### DISCUSSION THREAD

**Discussion by Navroza** [File-New-Project/EarTrumpet#648] [Ideas]
It would be great to add hotkeys for example I'd bind volume up/down and mute button of chrome on keyboard keys which I wont use so I could control music volume while gaming and I dont have to minimize and its time saver.

**Comment by clndwhr** [comment]
Has there been any movement with hotkeys for this?

**Comment by GoldenTao** [comment]
We're looking at a hotkey to have EarTrumpet appear for a future version, but we haven't really discussed specific app hotkeys, that could become quite problematic, but we'll look into it.

**Comment by tjcornish** [comment]
Thanks for a great and useful app.  I wanted to +1 this idea.  We are currently using PowerMixer which supports per-app keyboard shortcuts especially for muting/unmuting an app.  PowerMixer is not currently being developed and is problematic under Windows 10, so we're looking for a replacement.  Our use case is using shortcut keys to mute/unmute a TV stream when the user's phone rings or when someone walks into their office for a conversation.

Thanks

**Comment by yfmacit** [comment]
Please we need shortcut keys. Sometimes alt+tab can be a problem. Thanks to developers for this app.

**Comment by riverar** [comment]
We have a hotkey for opening the fly-out in EarTrumpet 2.0 (https://www.microsoft.com/en-us/p/eartrumpet/9nblggh516xp) but sounds like you want a few more. Will keep this open, bear with us.

**Reply by teadrinker2015** [reply]
That's a very comprehensive implementation out of my expectation. This answer should be marked at a more prominent position.

**Comment by ProH8R** [comment]
I would like to +1 this too. It would be especially useful when gaming.

For example when playing games, being able to quickly mute Chrome, Spotify, Music App etc without having to Alt Tab via F1-F12 or any other hotkey combination would be amazing. 

The Fly-Out menu is really great for games that support Full-Screen Windowed mode, but for games like Counter Strike Global Offensive, it causes a lengthy Alt-Tab.

Would love this to be added to the to-do list 👍

**Comment by WolfganP** [comment]
Adding my vote for this feature request. Currently my only nuisance with Windows is that the volume multimedia keys control the general/main volume. Having the ability to assign global shortcuts managed by EarTrumpet to volume per app (or even better: act on the currently streaming app, as usually is just one that outputs a constant stream -ie spotify, browser tab, local music/video player, ...-)

**Comment by Lordinio** [comment]
would love to be able to toggle between the audio devices or sources to change the volume with keyboard volume controls. If you could just choose between audio devices you wouldn't need to create new shortcuts for new programs

**Comment by infini667** [comment]
please guys, I beg you, make this possible! I'm suffering with this for a long time!! I want to control the spotify volume in game. these idiots could add this simple feature to spotify, but no... having to minimize the game everytime to control the volume is a pain. specially when I'm streaming. your software is great, just need this. thank you very much!

**Comment by andhemills** [comment]
My "solution" was to install 3rvx, then pin it to the taskbar. That way, it shows on the 2nd display's taskbar. I still can't get it to launch on the 2nd screen, but at least I can click on the taskbar and it makes 3rvx overlay over the [game] on my primary monitor, then I can adjust the sound.

I couldn't get EarTrumpet to work in a satisfactorily manner, since it's tied up to the quicklaunch, despite pinning it to the taskbar.
(I would like the pinned taskbar to launch EarTrumpet mixer).

Autohokey is a close second solution.
For Win+Page Up/Down:
`#PgUp::Send {Volume_Up 3}`
`#PgDn::Send {Volume_Down 3}`

Or I like Right-CTRL + Page Up/Down:
`>^PgUp::Send {Volume_Up 3}`
`>^PgDn::Send {Volume_Down 3}`

![image](https://user-images.githubusercontent.com/58420850/138543994-2f91bd15-0f13-46ca-a631-d14400872d38.png)

**Comment by rp1231** [comment]
Hey,
You can just use actions in the experimental build of eartrumpet to use hotkeys.

**Comment by ransworld** [comment]
Please add this feature to this already great program! I would love to be able to control the volume of music, game, and Discord independently with simple keystrokes.

**Comment by GBry** [comment]
It's been (almost) 8 years... Please?

**Reply by riverar** [reply]
@GBry Think it's been in the dev builds for that long too. Try one of those? Link at https://eartrumpet.app/

**Reply by kanjieater** [reply]
@riverar Thanks for pointing it out. You do have to dig pretty hard to find it. When can we expect it in a stable release?

**Reply by mage-anlyze** [reply]
@riverar Just a +1 thank you for this. The hotkey system is fantastic and offers a wealth of options. Thank you again.

**Reply by Globespy** [reply]
Wait....so, the hotkeys per app is a thing in the dev builds but not the official? Any specific build?
That's nuts it hasn't been added to the main version everyone downloads? 
It's a feature that literally makes the entire thing really worth using over the basic mixer

**Reply by riverar** [reply]
@Globespy Surprise!

**Reply by Globespy** [reply]
Awesome!! 
Just installed the dev app and found it! 
Fantastic!!
Thanks so much! 🙏
